test/performance/cicd/README.md

```diff
@@ -0,0 +1,50 @@
+# Kubestellar performance regression testing
+This is a simple performance test that aims to identify major regressions. 
+
+The setup is a single bindingpolicy and a single deployment. The test adds the deployment to the WDS and measures the amount of time it takes to create the WDS object, the Binding object, the ManifestWork object, and the deployment object on the WEC. The starting time is just prior to adding the WDS object and the end time is just after the deployment object is observed on the WEC. The other timings are obtained from the objects creation timestamp. The test intentionally looks at seconds rather than milliseconds since we expect this test to run on VMs and anything below seconds is likely just noise.
+
+This is what the results look like:
+```shell
+  Run 0: wds deployment=0, binding=1, manifestwork=1, wec deployment=1, total=2
+  Run 1: wds deployment=0, binding=0, manifestwork=0, wec deployment=5, total=6
+  Run 2: wds deployment=0, binding=1, manifestwork=1, wec deployment=7, total=7
+  Run 3: wds deployment=0, binding=0, manifestwork=0, wec deployment=5, total=5
+  Run 4: wds deployment=0, binding=0, manifestwork=0, wec deployment=4, total=5
+  Run 5: wds deployment=0, binding=0, manifestwork=0, wec deployment=4, total=5
+  Run 6: wds deployment=0, binding=0, manifestwork=0, wec deployment=4, total=5
+  Run 7: wds deployment=0, binding=1, manifestwork=1, wec deployment=6, total=7
+  Run 8: wds deployment=0, binding=0, manifestwork=0, wec deployment=4, total=5
+  Run 9: wds deployment=0, binding=0, manifestwork=0, wec deployment=5, total=5
+  ----------------------------------------------------------------------------------
+  Avg:   wds deployment=0, binding=0, manifestwork=0, wec deployment=4, total=5
+```
+
+## Running the tests
+To install Ginkgo, follow the instructions in [Ginkgo's getting started](https://onsi.github.io/ginkgo/#getting-started).
+
+To execute these tests, issue the following command. 
```

Comment:
So this test is not only for CI, a contributor can run it too. I would not put this test in a "cicd" subdirectory.

Reply:
Agree

Reply:
Would regression be a better fit?

Reply:
I think that using the word "performance" rather than "regression" is appropriate, since other sorts of regressions are known (e.g., functional regression).

Since this is an end-to-end test, I would be happy with it appearing in `test/e2e/performance/`.

Reply:
I think of this test as more more than just an e2e test. It intentionally tracks the amount of time it takes to create the binding.control, manifestwork, and finally the deployment into wecs.  The idea behind it is that if someone introduces a deplay into any of these stages we can tell at a glance using a relatively simple test.

Reply:
I think of end-to-end to be in a different dimension than performance vs functionality. We can test performance in end-to-end scenarios and test performance in smaller scopes. We can test functionality in end-to-end scenarios and test functionality in smaller scopes.

We can have functionality regressions and we can have performance regressions. Looking at the first paragraph in https://en.wikipedia.org/wiki/Regression_testing, it seems to be saying that "regression testing" is another term for what we call Continuous Integration. I do not think that we need the word "regression" in the pathname for these new tests, any more than we need it in the pathname for the tests already merged.

Not a blocker.